modules/drtracker.lua

```diff
@@ -194,64 +248,84 @@ function DRTracker:DRFaded(unit, spellID)
-				frame:SetPoint("TOPLEFT", self.frame[unit])
-			elseif self.db[unit].drTrackerGrowDirection == "RIGHT" then
-				frame:SetPoint("TOPLEFT", lastFrame, "TOPRIGHT", self.db[unit].drTrackerMargin, 0)
-			elseif self.db[unit].drTrackerGrowDirection == "LEFT" then
-				frame:SetPoint("TOPRIGHT", lastFrame, "TOPLEFT", -self.db[unit].drTrackerMargin, 0)
-			elseif self.db[unit].drTrackerGrowDirection == "UP" then
-				frame:SetPoint("BOTTOMLEFT", lastFrame, "TOPLEFT", 0, self.db[unit].drTrackerMargin)
-			elseif self.db[unit].drTrackerGrowDirection == "DOWN" then
-				frame:SetPoint("TOPLEFT", lastFrame, "BOTTOMLEFT", 0, -self.db[unit].drTrackerMargin)
-			end
-
-			lastFrame = frame
-
-			frame:Show()
-		else
-			frame:Hide()
-		end
-	end
-end
-
 function DRTracker:COMBAT_LOG_EVENT_UNFILTERED(event)
 	self:CombatLogEvent(event, CombatLogGetCurrentEventInfo())
 end
 
-function DRTracker:CombatLogEvent(event, timestamp, eventType, hideCaster, sourceGUID, sourceName, sourceFlags, sourceRaidFlags, destGUID, destName, destFlags, destRaidFlags, spellID, spellName, spellSchool, auraType)
+function DRTracker:CombatLogEvent(_, timestamp, eventType, hideCaster, sourceGUID, sourceName, sourceFlags, sourceRaidFlags, destGUID, destName, destFlags, destRaidFlags, spellID, spellName, spellSchool, auraType)
 	-- Enemy had a debuff refreshed before it faded
 	-- Buff or debuff faded from an enemy
-	if eventType == "SPELL_AURA_REFRESH" or eventType == "SPELL_AURA_REMOVED" then
-		if auraType == "DEBUFF" and DRData:GetSpellCategory(spellID) then
+	if eventType == "SPELL_AURA_APPLIED" or eventType == "SPELL_AURA_REFRESH" or eventType == "SPELL_AURA_REMOVED" then
+		local drCat, catTbl = DRData:GetSpellCategory(spellID)
+		if drCat and auraType == "DEBUFF" then
 			local unit = GladiusEx:GetUnitIdByGUID(destGUID)
 			if unit and self.frame[unit] then
-				self:DRFaded(unit, spellID)
+				local tbl = catTbl and catTbl or { drCat }
+				for _, drCat in pairs(tbl) do -- K: Use a for loop to catch auras that DR with multiple categories (primarily Ring of Frost/Deep Freeze)				
+					
+					-- K: Dynamic DR reset early if we have a full duration aura on _REFRESH / _APPLIED 
+					local tracked = (self.frame[unit] and self.frame[unit].tracker) and self.frame[unit].tracker[drCat] or nil
+					if tracked and (eventType == "SPELL_AURA_REFRESH" or eventType == "SPELL_AURA_APPLIED") then
+						if self:HasFullDurationAura(unit, sourceGUID, spellID) then
+							tracked.active = false
+						end
+					end
+					
+					self:DRFaded(unit, drCat, spellID, eventType)
+				end
+			end
+		end
+	end
+end
+
+function DRTracker:HasFullDurationAura(unit, sourceGUID, spellID)
+	local fullDuration = GladiusEx.Data.AuraDurations and GladiusEx.Data.AuraDurations[spellID] or nil
+	
+	if fullDuration then
+		local srcUnit = GladiusEx:GetUnitIdByGUID(sourceGUID)
+
+		for i=1, 40 do

```

Comment:
This shouldn’t be up to 40, it should just keep its `break` exit condition

Reply:
40 is the maximum number of harmful auras one can have. This is how all addons tracking auras do it. Including this one, see line 124 of classicon.lua.

Reply:
Then it's a bug in the addon. That limit only exists on classic and has been removed on retail AFAIK.

Reply:
WeakAuras uses the same approach but 255 instead of 40 (probably because it doesn't make a difference and future proofing to 255 isn't a bad idea), and most other AddOns I've checked use 40.

I can change it to 255 but I don't see the point in completely changing the structure to a repeat until loop.

Also, while I don't have any evidence, I am fairly sure 40 is the limit also in retail (at least I am finding posts on wowinterface from ~2018 suggesting it was so back then).

Reply:
Wrath-Logo-Small.png [Patch 3.0.2](https://warcraft.wiki.gg/wiki/Patch_3.0.2) (2008-10-14): There is no longer a limit on the amount of debuffs a target can have on them at any time.

^ You are correct. I will modify accordingly.